### PR TITLE
[Bug 1883593] Instrumentation to capture more clicks via glean

### PR DESCRIPTION
**Pull request by abhi-agg**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1883593
### Pull Request checklist

- [ ] The pull request has a descriptive title (and a reference to an issue it
      fixes, if applicable)
- [ ] All tests and linter checks are passing
- [ ] The pull request is free of merge conflicts

### CODE REVIEW

File: src/components/AuthenticatedLink.svelte [hunk 1]
```diff
@@ -2,9 +2,11 @@
   import tippy from "./tippy";
 
   export let href;
+  export let label = null; // a prop for glean click events
+  export let type = null; // a prop for glean click events
 </script>
 
-<a {href}><slot /></a>
+<a {href} data-glean-label={label} data-glean-type={type}><slot /></a>
```
Comment (rosahbruno) [inline src/components/AuthenticatedLink.svelte:9]: I am not 100% sure how Svelte works with props passing. But, if there is a way to only set those properties if we are guaranteed to have a value for them (so they aren't incorrectly set as undefined or something) then that would be good. Unless Glean.js is checking to make sure those values aren't undefined? I forget how the Glean.js implementation looks

**Comment by abhi-agg** [comment]
Okay. I am marking this PR read to review. This PR adds instrumentation to capture clicks on:

1. Metric details page: Clicks on `Access` section (bottom most) and `Sampling Configuration Snippet ` row inside `Metric sampling` section

2. App details page: Clicks on `Metrics`, `Pings`, `Tags`, `Application IDs` and `Third-Party Data` tabs

I have tested this change locally and it works as expected.

File: src/pages/AppDetail.svelte [hunk 2]
```diff
@@ -91,15 +91,17 @@
...
+      <Tab key="tags" label="AppDetail.TagsTab">Tags</Tab>
     {/if}
-    <Tab key="app_ids">Application IDs</Tab>
+    <Tab key="app_ids" label="AppDetail.ApplicationIdsTab">Application IDs</Tab>
     {#if app.app_name === "firefox_ios"}
       <!-- for now, only Firefox iOS uses third-party data collection -->
-      <Tab key="third_party_data">Third-Party Data</Tab>
+      <Tab key="third_party_data" label="AppDetail.ThirdPartyDataTab"
+        >Third-Party Data</Tab
+      >
```
Comment (travis79) [inline src/pages/AppDetail.svelte:104]: Formatting here looks weird, is this what the linter/formatter wanted?

**Review by travis79 — APPROVED** [review]
Looks good to me, with one minor question/nit about formatting in one place.

File: src/components/AuthenticatedLink.svelte (hunk 1 above)
Comment (rosahbruno) [reply src/components/AuthenticatedLink.svelte:9]: Yep, sounds good 👍

File: src/pages/AppDetail.svelte (hunk 2 above)
Comment (abhi-agg) [reply src/pages/AppDetail.svelte:104]: Yeah. This is the final output of running formatter and linter.

**Pull request merged**